controllers/argocd_controller.go

```diff
@@ -119,26 +132,32 @@ func (r *ReconcileArgoCDRoute) Reconcile(ctx context.Context, request reconcile.
 		}
 		return reconcile.Result{}, err
 	}
+
 	reqLogger.Info("Route found for argocd-server", "Route.Host", argoCDRoute.Spec.Host)
 
-	argocCDRouteURL := fmt.Sprintf("https://%s", argoCDRoute.Spec.Host)
+	argoCDRouteURL := fmt.Sprintf("https://%s", argoCDRoute.Spec.Host)
 
-	consoleLink := newConsoleLink(argocCDRouteURL, "Cluster Argo CD")
+	consoleLink := newConsoleLink(argoCDRouteURL, "Cluster Argo CD")
 
 	found := &console.ConsoleLink{}
 	err = r.Client.Get(ctx, types.NamespacedName{Name: consoleLink.Name}, found)
-	if err != nil {
-		if errors.IsNotFound(err) {
-			reqLogger.Info("Creating a new ConsoleLink", "ConsoleLink.Name", consoleLink.Name)
-			return reconcile.Result{}, r.Client.Create(ctx, consoleLink)
+	// if env variable for DISABLE_DEFAULT_ARGOCD_CONSOLELINK is "true", delete the Argo CD ConsoleLink or don't create it
+	if isConsoleLinkDisabled() {
+		if err != nil {
+			if errors.IsNotFound(err) {
+				reqLogger.Info("Creating a new ConsoleLink", "ConsoleLink.Name", consoleLink.Name)
+				return reconcile.Result{}, r.Client.Create(ctx, consoleLink)
+			}
+			reqLogger.Error(err, "Failed to create ConsoleLink", "ConsoleLink.Name", consoleLink.Name)
+			return reconcile.Result{}, err
 		}
-		reqLogger.Error(err, "Failed to create ConsoleLink", "ConsoleLink.Name", consoleLink.Name)
-		return reconcile.Result{}, err
+		reqLogger.Info("consoleLink deleted", "Route.Host", argoCDRoute.Spec.Host)
```

Comment:
This "consoleLink deleted" will get printed repeatedly even when console link does not exist.

Reply:
We have to prevent "consoleLink deleted" message from printing when there is no consoleLink to delete.

Reply:
We can remove the Info logging here as deleteConsoleLinkIfPresent() already writes to the log when a link is deleted.